addons/mail/static/src/models/message_action_list.js

```diff
@@ -1,177 +1,99 @@
-         */
-        _computeHasReplyIcon() {
-            return Boolean(
-                this.messaging && this.messaging.inbox &&
-                this.message && !this.message.isTemporary && !this.message.isTransient &&
-                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread && (
-                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox ||
-                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread.model === 'mail.channel'
-                )
-            );
+            ) {
+                messageActionViewsData.push({actionType: 'MarkAsRead'});
+            }
+            if (this.message.hasReactionIcon) {
+                messageActionViewsData.push({actionType: 'Reaction'});
+            }
+            if (this.hasReplyToAction) {
+                messageActionViewsData.push({actionType: 'ReplyTo'});
+            }
+            if (this.hasCompactAction) {
+                messageActionViewsData.push({actionType: 'ToggleCompact'});
+            }
+            if (this.message.canStarBeToggled) {
+                messageActionViewsData.push({actionType: 'ToggleStar'});
+            }
+            return insertAndReplace(messageActionViewsData);
         },
+        _sortMessageActionViews() {
+            return [
+                ['smaller-first', 'sequence'],
+            ];
+        }
     },
     fields: {
-        /**
-         * States the reference to the reaction action in the component.
-         */
-        actionReactionRef: attr(),
         actionsCount: attr({
-            compute: '_computeActionsCount',
             readonly: true,
+            sum: 'messageActionViews',
         }),
-        addReactionText: attr({
-            compute: '_computeAddReactionText',
-        }),
-        deleteConfirmDialog: one('Dialog', {
-            inverse: 'messageActionListOwnerAsDeleteConfirm',
-            isCausal: true,
+        actionsWithoutCompactCount: attr({
+            compute: '_computeActionViewsWithoutCompactCount',
+            readonly: true,
         }),
-        hasCompactIcon: attr({
-            compute: '_computeHasCompactIcon',
+        compactThreshold: attr({
+            default: 2,
             readonly: true,
         }),
-        /**
-         * Determines whether this message action list has mark as read icon.
-         */
-        hasMarkAsReadIcon: attr({
-            compute: '_computeHasMarkAsReadIcon',
+        hasCompactAction: attr({
+            compute: '_computeHasCompactAction',
         }),
-        /**
-         * Determines whether this message action list has a reply icon.
-         */
-        hasReplyIcon: attr({
-            compute: '_computeHasReplyIcon',
+        hasReplyToAction: attr({
```

Comment:
example:

`replyToActionView`: `_computeReplyToActionView`
inverse: 'messageActionListOwnerAsReplyTo`

inside `MessageActionView`
compute: `messageActionListOwner`, inverse: `messageActionViews` (existing field here, but computed from the inverse)